File: Practica 2 - Java RMI/GestorBibliotecaService/src/gestor/biblioteca/service/models/TLibro.java
package gestor.biblioteca.service.models;

import java.io.Serializable;
import java.util.Objects;

/**
 *
 * @author Saúl Rodríguez Naranjo
 */
public class TLibro implements Serializable, Cloneable
{
    private String titulo;
    private String autor;
    private String pais;
    private String idioma;
    private String isbn;
    private int anio;
    
    private int disponibles;
    private int prestados;
    private int reservados;

    public TLibro()
    {
    }

    public TLibro(String titulo, String autor, String pais, String idioma, 
            String isbn, int anio, int disponibles, int prestados, 
            int reservados)
    {
        this.titulo = titulo;
        this.autor = autor;
        this.pais = pais;
        this.idioma = idioma;
        this.isbn = isbn;
        this.anio = anio;
        this.disponibles = disponibles;
        this.prestados = prestados;
        this.reservados = reservados;
    }

    public String getTitulo()
    {
        return titulo;
    }

    public void setTitulo(String titulo)
    {
        this.titulo = titulo;
    }

    public String getAutor()
    {
        return autor;
    }

    public void setAutor(String autor)
    {
        this.autor = autor;
    }

    public String getPais()
    {
        return pais;
    }

    public void setPais(String pais)
    {
        this.pais = pais;
    }

    public String getIdioma()
    {
        return idioma;
    }

    public void setIdioma(String idioma)
    {
        this.idioma = idioma;
    }

    public String getIsbn()
    {
        return isbn;
    }

    public void setIsbn(String isbn)
    {
        this.isbn = isbn;
    }

    public int getAnio()
    {
        return anio;
    }

    public void setAnio(int anio)
    {
        this.anio = anio;
    }

    public int getDisponibles()
    {
        return disponibles;
    }

    public void setDisponibles(int disponibles)
    {
        this.disponibles = disponibles;
    }

    public int getPrestados()
    {
        return prestados;
    }

    public void setPrestados(int prestados)
    {
        this.prestados = prestados;
    }

    public int getReservados()
    {
        return reservados;
    }

    public void setReservados(int reservados)
    {
        this.reservados = reservados;
    }

    @Override
    public int hashCode()
    {
        int hash = 3;
        hash = 37 * hash + Objects.hashCode(this.titulo);
        hash = 37 * hash + Objects.hashCode(this.autor);
        hash = 37 * hash + Objects.hashCode(this.pais);
        hash = 37 * hash + Objects.hashCode(this.idioma);
        hash = 37 * hash + Objects.hashCode(this.isbn);
        hash = 37 * hash + Objects.hashCode(this.anio);
        hash = 37 * hash + this.disponibles;
        hash = 37 * hash + this.prestados;
        hash = 37 * hash + this.reservados;
        return hash;
    }

    @Override
    public boolean equals(Object obj)
    {
        if (this == obj)
        {
            return true;
        }
        if (obj == null)
        {
            return false;
        }
        if (getClass() != obj.getClass())
        {
            return false;
        }
        final TLibro other = (TLibro) obj;
        if (this.disponibles != other.disponibles)
        {
            return false;
        }
        if (this.prestados != other.prestados)
        {
            return false;
        }
        if (this.reservados != other.reservados)
        {
            return false;
        }
        if (!Objects.equals(this.titulo, other.titulo))
        {
            return false;
        }
        if (!Objects.equals(this.autor, other.autor))
        {
            return false;
        }
        if (!Objects.equals(this.pais, other.pais))
        {
            return false;
        }
        if (!Objects.equals(this.idioma, other.idioma))
        {
            return false;
        }
        if (!Objects.equals(this.isbn, other.isbn))
        {
            return false;
        }
        return Objects.equals(this.anio, other.anio);
    }

    @Override
    public String toString()
    {
        StringBuilder sb = new StringBuilder();
        sb.append("TLibro{");
        sb.append("titulo=").append(titulo);
        sb.append(", autor=").append(autor);
        sb.append(", pais=").append(pais);
        sb.append(", idioma=").append(idioma);
        sb.append(", isbn=").append(isbn);
        sb.append(", anio=").append(anio);
        sb.append(", disponibles=").append(disponibles);
        sb.append(", prestados=").append(prestados);
        sb.append(", reservados=").append(reservados);
        sb.append('}');
        return sb.toString();
    }

    @Override
    public Object clone() throws CloneNotSupportedException
    {
        TLibro clone = new TLibro(titulo, autor, pais, idioma, isbn, anio, 
                disponibles, prestados, reservados);
        
        return clone;
    }
    
    
    
}

File: Practica 3 - C# Net Remoting/GestorBibliotecaService/GestorBibliotecaService/TLibro.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GestorBibliotecaService
{
    [Serializable]
    public class TLibro : IEquatable<TLibro>, ICloneable
    {
        private string titulo;
        private string autor;
        private string pais;
        private string idioma;
        private string isbn;
        private int anio;

        private int disponibles;
        private int prestados;
        private int reservados;

        public TLibro()
        {
        }

        public TLibro(string titulo, string autor, string pais, string idioma,
                string isbn, int anio, int disponibles, int prestados,
                int reservados)
        {
            this.titulo = titulo;
            this.autor = autor;
            this.pais = pais;
            this.idioma = idioma;
            this.isbn = isbn;
            this.anio = anio;
            this.disponibles = disponibles;
            this.prestados = prestados;
            this.reservados = reservados;
        }

        public string Titulo
        {
            get { return this.titulo; }
            set { this.titulo = value; }
        }

        public string Autor
        {
            get { return this.autor; }
            set { this.autor = value; }
        }

        public string Pais
        {
            get { return this.pais; }
            set { this.pais = value; }
        }

        public string Idioma
        {
            get { return this.idioma; }
            set { this.idioma = value; }
        }

        public string Isbn
        {
            get { return this.isbn; }
            set { this.isbn = value; }
        }

        public int Anio
        {
            get { return this.anio; }
            set { this.anio = value; }
        }

        public int Disponibles
        {
            get { return this.disponibles; }
            set { this.disponibles = value; }
        }

        public int Prestados
        {
            get { return this.prestados; }
            set { this.prestados = value; }
        }

        public int Reservados
        {
            get { return this.reservados; }
            set { this.reservados = value; }
        }

        public object Clone()
        {
            return this.MemberwiseClone();
        }

        public override bool Equals(object obj)
        {
            return Equals(obj as TLibro);
        }

        public bool Equals(TLibro other)
        {
            return other != null &&
                   isbn == other.isbn;
        }

        public override int GetHashCode()
        {
            return 1553653001 + EqualityComparer<string>.Default.GetHashCode(isbn);
        }

        public static bool operator ==(TLibro left, TLibro right)
        {
            return EqualityComparer<TLibro>.Default.Equals(left, right);
        }

        public static bool operator !=(TLibro left, TLibro right)
        {
            return !(left == right);
        }

        public override string ToString()
        {
            string s = "{" + this.titulo + ", " + this.autor + ", " + this.anio + ", "
                + this.pais + ", " + this.idioma + ", " + this.isbn + ", " + this.disponibles
                + ", " + this.prestados + ", " + this.reservados + "}";

            return s;
        }
    }
}
